src/main/cpp/locationinfo.cpp

```diff
@@ -35,6 +41,21 @@ const LocationInfo& LocationInfo::getLocationUnavailable()
 	return unavailable;
 }
 
+/**
+ * filter short file name
+ * @param fileName
+ * @return shortFileName
+ */
+const std::string filterFileName(const char* const fileName){
+	std::string fullFileName(fileName);
+	std::size_t found = fullFileName.rfind(SHORT_FILENAME_SPLIT_CHAR);
+	if (found != std::string::npos) {
+		return fullFileName.substr(found + 1);
+	} else {
+		return std::string(fileName);
+	}
```

Comment:
As per Thorsten's comments, this should probably do more of a `const char*` thing.  We can be somewhat clever here - since we know that the actual filename is going to be at the end of the filename, we just need to walk the string until we get to the end.  Something like this should work:

```
const char* begin = fileName;
size_t pos = 0;
while( fileName[pos] != 0 ){
  if( fileName[pos] == SHORT_FILENAME_SPLIT_CHAR &&
      fileName[pos+1] != 0 ){
      begin = fileName + pos + 1;
  }
  pos++;
}
return begin;
```

This also means that we don't _have_ to add a new member variable(this could be the implementation of `getShortFileName`), but adding a new member variable is probably good(I think with C++17 we can actually make this `constexpr` to evaluate at compile-time, but we would need to add a check for that).

Reply:
If C++17 is an option, [basic_string_view::rfind](https://en.cppreference.com/w/cpp/string/basic_string_view/rfind) could be used. Avoiding `std::string::rfind` doesn't seem that important to me, but if so, the look should be backwards. Something like the following:

```cpp
size_t len = strlen(fileName);

for (size_t pos = std::max(len - 1, 0); pos >= 0; --pos)
{
  if (fileName[pos] == SHORT_FILENAME_SPLIT_CHAR)
  {
    return fileName[pos + 1];
  }

  if (pos == 0)
  {
    return fileName;
  }
}
```